src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/CohortController.cs

```diff
@@ -226,13 +226,39 @@ public async Task<IActionResult> AddDraftApprenticeshipOrRoute(string changeCour
             if (await HasStartDateOverlap(model))
             {
                 StoreDraftApprenticeshipState(model);
-                return RedirectToAction(nameof(DraftApprenticeshipOverlapOptions));
+                return RedirectToAction(nameof(DraftApprenticeshipOverlapAlert));
             }
 
             return await SaveDraftApprenticeship(model);
         }
 
-          
+        [HttpGet]
+        [Route("add/apprenticeship/overlap-alert")]
+        public IActionResult DraftApprenticeshipOverlapAlert(DraftApprenticeshipOverlapAlertRequest request, [FromServices] IFeatureTogglesService<ProviderFeatureToggle> featureTogglesService)
+        {
+            var featureToggleEnabled = featureTogglesService.GetFeatureToggle(ProviderFeature.OverlappingTrainingDate).IsEnabled;
+            var model = PeekStoredDraftApprenticeshipState();
+            
+            var vm = new DraftApprenticeshipOverlapAlertViewModel
+            {
+                OverlappingTrainingDateRequestToggleEnabled = featureToggleEnabled,
+                DraftApprenticeshipHashedId = request.DraftApprenticeshipHashedId,
+                StartDate = model is null ? DateTime.MinValue : model.StartDate.Date.GetValueOrDefault(),
+                EndDate = model is null ? DateTime.MaxValue : model.EndDate.Date.GetValueOrDefault(),
+                Uln = model is null ? string.Empty : model.Uln,
+                FirstName = model is null ? string.Empty : model.FirstName,
+                LastName = model is null ? string.Empty : model.LastName
```

Comment:
If the `model is null` here then something has gone wrong and we shouldn't be progressing to the next screen with a bunch of default values. None of the other controller actions perform this check. If it were to happen then we should navigate to the screen before this where we input apprenticeship details again.

Reply:
This has been updated. Unit tests updated accordingly.